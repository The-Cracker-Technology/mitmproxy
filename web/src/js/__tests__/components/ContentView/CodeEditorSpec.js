jest.mock('react-codemirror')
import React from 'react'
import renderer from 'react-test-renderer'
import CodeEditor from '../../../components/ContentView/CodeEditor'

describe('CodeEditor Component', () => {
    let content = "foo content",
        changeFn = jest.fn(),
        codeEditor = renderer.create(
            <CodeEditor content={content} onChange={changeFn}/>
        ),
        tree = codeEditor.toJSON()
    
    it('should render correctly', () => {
        // This actually does not render properly, but getting a full CodeMirror rendering
        // is cumbersome. This is hopefully good enough.
        // see: https://github.com/mitmproxy/mitmproxy/pull/2365#discussion_r119766850
        expect(tree).toMatchSnapshot()
    })

    it('should handle key down', () => {
        let mockEvent = { stopPropagation: jest.fn() }
        tree.props.onKeyDown(mockEvent)
        expect(mockEvent.stopPropagation).toBeCalled()
    })
})
