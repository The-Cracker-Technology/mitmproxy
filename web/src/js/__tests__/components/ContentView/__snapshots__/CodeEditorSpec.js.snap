// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodeEditor Component should render correctly 1`] = `
<div
  className="codeeditor"
  onKeyDown={[Function]}
/>
`;
